internal/config/device/device_test.go

```diff
@@ -0,0 +1,48 @@
+package device_test
+
+import (
+	"github.com/cri-o/cri-o/internal/config/device"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = t.Describe("DeviceConfig", func() {
+	var d *device.Config
+	BeforeEach(func() {
+		d = device.New()
+	})
+	t.Describe("LoadDevices", func() {
+		It("should fail with poorly formatted device", func() {
+			// Given
+			// When
+			err := d.LoadDevices([]string{"invalid:invalid"})
+			// Then
+			Expect(err).NotTo(BeNil())
+			Expect(d.Devices()).To(BeEmpty())
+		})
+		It("should fail if invalid device", func() {
+			// Given
+			// When
+			err := d.LoadDevices([]string{"/dev/invalid"})
+			// Then
+			Expect(err).NotTo(BeNil())
+			Expect(d.Devices()).To(BeEmpty())
+		})
+		It("should succeed with valid device", func() {
+			// Given
+			// When
+			err := d.LoadDevices([]string{"/dev/null:/dev/null:w"})
```

Comment:
Maybe I'll discover later down in the code, but is a spec like `"/dev/null::w"` valid?  Regardless you probably want to add a test here for it to make sure it's handled.

Reply:
WOW thanks apparently we segfault :upside_down_face:

Reply:
fixed in https://github.com/containers/podman/pull/8200